BlueprintUICommonControls/Sources/UserInteractionEnabled.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  UserInteractionEnabled.swift
+//  BlueprintUICommonControls
+//
+//  Created by Noah Blake on 3/15/21.
+//
+
+import BlueprintUI
+import UIKit
+
+/// `UserInteractionEnabled` conditionally enables user interaction of its wrapped element.
+///
+/// - Note: When user interaction is disabled, any elements within the wrapped element will become non-interactive.
+public struct UserInteractionEnabled: Element {
+    public var isEnabled: Bool
+    public var wrappedElement: Element
+
+    public init(_ isEnabled: Bool, wrapping element: Element) {
+        self.isEnabled = isEnabled
+        self.wrappedElement = element
+    }
+
+    public var content: ElementContent {
+        ElementContent(child: wrappedElement)
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        UIView.describe { config in
+            config[\.isUserInteractionEnabled] = isEnabled
+        }
+    }
+}
+
+public extension Element {
+    func isUserInteractionEnabled(_ isEnabled: Bool) -> UserInteractionEnabled {
```

Comment:
Bikeshedding, but I would probably drop the `is` prefix on this modifier. SwiftUI just calls this [`disabled`](https://developer.apple.com/documentation/swiftui/view/disabled(_:)), if you want to go that route.

Also, could you add a doc comment to this? You can just copy what the `UserInteractionEnabled` type comment says.

Reply:
Updated to `disabled` and add docs.

```swift
func disabled(_ disabled: Bool = true) -> UserInteractionEnabled { ... }
```

Reply:
Does the SwiftUI version also change any visual appearances? Since we dont here, I honestly prefer the `userInteractionEnabled` name we had before, but not a big deal (eg, thinking about how on a traditional UIKit control, setting `isEnabled = false` usually also affects the visual appearance, but we dont do that here).

Reply:
> Does the SwiftUI version also change any visual appearances

Oh, good catch, these are not the same thing. `userInteractionEnabled` is probably better in that case.

([allowsHitTesting](https://developer.apple.com/documentation/swiftui/list/allowshittesting(_:)) might be the next closest thing but I didn't test it)

Reply:
Renamed to `userInteractionEnabled(_ enabled: Bool = true) -> UserInteractionEnabled`. Thanks for the close read here.